extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/telemetry/TracesConnectionInterceptor.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.websockets.next.runtime.telemetry;
+
+import static io.quarkus.websockets.next.runtime.telemetry.TelemetryConstants.CONNECTION_FAILURE_ATTR_KEY;
+import static io.quarkus.websockets.next.runtime.telemetry.TelemetryConstants.URI_ATTR_KEY;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.Tracer;
+
+final class TracesConnectionInterceptor implements ConnectionInterceptor {
+
+    static final String CONNECTION_OPENED_SPAN_CTX = "io.quarkus.websockets.next.connection-opened-span-ctx";
+
+    private final Tracer tracer;
+    private final String connectionOpenedSpanName;
+    private final String connectionOpeningFailedSpanName;
+    private final String path;
+    private final Map<String, Object> contextData;
+
+    TracesConnectionInterceptor(Tracer tracer, String connectionOpenedSpanName, String connectionOpeningFailedSpanName,
+            String path) {
+        this.tracer = tracer;
+        this.connectionOpenedSpanName = connectionOpenedSpanName;
+        this.connectionOpeningFailedSpanName = connectionOpeningFailedSpanName;
+        this.path = path;
+        this.contextData = new HashMap<>();
+    }
+
+    @Override
+    public void connectionOpened() {
+        var span = tracer.spanBuilder(connectionOpenedSpanName)
```

Comment:
@brunobat I can read the code wrong, but this is how I read it:

- making span means `ContextStorage.get().attach(this)`
- attaching storage invokes `io.quarkus.opentelemetry.runtime.QuarkusContextStorage#attach(io.opentelemetry.context.Context)`
- local data are added to the duplicated context that are not there yet
- `io.vertx.core.impl.ContextInternal#LOCAL_MAP` relies on concurrent map
- I once tried to use these local data when I didn't have to and here I got some context on performance impact it can have https://github.com/quarkusio/quarkus/pull/37795#issuecomment-1860666461

I don't mind doing that, but I believe we should get something if we pay a price. When you say safer, is this "safer" important? I don't know details so I cannot decide that, I just want to be sure this is right thing to do.

Reply:
Safer because this line might throw an (very unlikely) error: `contextData.put(CONNECTION_OPENED_SPAN_CTX, span.getSpanContext());` and we want to close the span regardless. This is the spec recommendation.

Reply:
Alright, I'll do that.